documentation/modules/configuring/con-config-mirrormaker2-producers-consumers.adoc

```diff
@@ -0,0 +1,109 @@
+// Module included in the following assemblies:
+//
+// assembly-config-mirrormaker2.adoc
+
+[id='con-mirrormaker-producers-consumers-{context}']
+= Connector producer and consumer configuration
+
+[role="_abstract"]
+MirrorMaker 2.0 connectors uses internal producers and consumers.
+If needed, you can configure these producers and consumers to override the default settings. 
+
+For example, you can increase the `batch.size` for the source producer that sends topics to the target Kafka cluster to better accommodate large volumes of messages.
+
+IMPORTANT: Producer and consumer configuration options depend on the MirrorMaker 2.0 implementation, and may be subject to change.  
+
+The following tables describe the producers and consumers for each of the connectors and where you can add configuration. 
+
+.Source connector producers and consumers
+[cols="1,1a,3m",options="header"]
+|===
+
+|Type
+|Description
+|Configuration
+
+|Producer
+|Sends topic messages to the target Kafka cluster. Consider tuning the configuration of this producer when it is handling large volumes of data. 
+|mirrors.sourceConnector.config: producer.override.*
+
+|Producer
+|Writes to the `offset-syncs` topic, which maps the source and target offsets for replicated topic partitions. 
+|mirrors.sourceConnector.config: producer.*
+
+|Consumer
+|Retrieves topic messages from the source Kafka cluster.
+|mirrors.sourceConnector.config: consumer.* 
+|===
+
+.Checkpoint connector producers and consumers
+[cols="1,1a,3m",options="header"]
+|===
+
+|Type
+|Description
+|Configuration
+
+|Producer
+|Emits consumer offset checkpoints.
+|mirrors.checkpointConnector.config: producer.override.* 
+
+|Consumer
+|Loads the `offset-syncs` topic.
+|mirrors.checkpointConnector.config: consumer.*
+
+|===
+
+IMPORTANT: The consumer configuration changes depending on the location of the `offset-syncs` topic. The default location is the source Kafka cluster. You can set `offset-syncs.topic.location` to `target` to use the target Kafka cluster. If you use the target Kafka cluster, use `mirrors.checkpointConnector.config: target.consumer.*` for the consumer configuration. 
```

Comment:
I think we can remove this sentence. The following is the same as described in [ENTMQST-4116](https://issues.redhat.com/browse/ENTMQST-4116): 
```
I think the users should use only `mirrors.checkpointConnector.config: consumer.*` to configure the consumer in MirrorCheckpointConnector, and they should not use {source_or_target} prefix.
There is no other consumers in checkpointConnector, and it would be misleading to use different source and target depending on the value of offset-syncs.topic.location, so it is not necessary to specify further whether it is source or target cluster.

I think it is sufficient to explain that only `mirrors.checkpointConnector.config: consumer.*` is used to configure the consumer in MirrorCheckpointConnector.
```

Reply:
I thought it might be useful to mention `offset-syncs.topic.location` here, so changed to:

> NOTE: You can set `offset-syncs.topic.location` to `target` to use the target Kafka cluster as the location of the `offset-syncs` topic.